relational_operators/WorkOrderFactory.cpp

```diff
@@ -113,7 +113,8 @@ WorkOrder* WorkOrderFactory::ReconstructFromProto(const serialization::WorkOrder
     case serialization::DESTROY_HASH: {
       LOG(INFO) << "Creating DestroyHashWorkOrder";
       return new DestroyHashWorkOrder(
-          proto.GetExtension(serialization::DestroyHashWorkOrder::join_hash_table_index),
+          *(catalog_database->getRelationById(proto.GetExtension(serialization::DestroyHashWorkOrder::relation_id))),
```

Comment:
We need `PartitionSchemeHeader`, not `CatalogRelation`. Serialize it in `WorkOrder` proto.